app/views/settings/dex_connector_oidcs/_form.html.slim

```diff
@@ -0,0 +1,70 @@
+  = f.password_field :password, style: "display:none", value:"autocomplete bait"
+
+
+  .form-group class="#{error_class_for(@data_holder, :name)}"
+    = f.label :name
+    = f.text_field :name, id: "oidc_name", class: "form-control", value: @data_holder.name, required: true
+    = error_messages_for(@data_holder, :name)
+    small.form-text.text-muted
+      | Name shown to user when selecting a connector
+
+  h4 Provider
+
+  .form-group class="#{error_class_for(@data_holder, :provider_url)}"
+    = f.label :provider_url, "Provider URL"
+    = f.text_field :provider_url, id: "oidc_provider", class: "form-control", value: @data_holder.provider_url, required: true
+    = error_messages_for(@data_holder, :provider_url)
+    small.form-text.text-muted
+      | Issuer used by OIDC Provider
+    
+  .form-group class="#{error_class_for(@data_holder, :basic_auth)}"
+    = f.label :basic_auth, "Basic Auth"
+    br
+    .btn-group.btn-group-toggle data-toggle="buttons"
+      = f.label :basic_auth, nil, class: "btn btn-default #{'btn-primary active' if @data_holder.basic_auth}"
+        = f.radio_button :basic_auth, "true", checked: @data_holder.basic_auth
+        | On
+      = f.label :basic_auth, nil, class: "btn btn-default #{'btn-primary active' unless @data_holder.basic_auth}"
+        = f.radio_button :basic_auth, "false", checked: !@data_holder.basic_auth
+        | Off
+    = error_messages_for(@data_holder, :basic_auth)
+    br
+    small.form-text.text-muted
+      | Set to "Off" only if this OIDC provider requires credentials to be sent using GET (this is uncommon)
+    
+  h4 Client
+  
+  / The 'https' and '3200' seem to be hard-coded elsewhere, so I assume they're not discoverable :/
+  - external_api_url = "https://#{Pillar.value(pillar: :apiserver)}:32000/callback"
+  .form-group class="#{error_class_for(@data_holder, :callback_url)}"
+    = f.label :callback_url, "Callback URL:"
+    span.show
+      = external_api_url
+    small.form-text.text-muted
+      | URL which will be provided to OIDC provider as the callback location
+    = f.hidden_field :callback_url, id: "oidc_callback_url", class: "form-control", value: external_api_url, readonly: true
+
+  .form-group class="#{error_class_for(@data_holder, :client_id)}"
+    = f.label :client_id, "ID"
+    = f.text_field :client_id, id: "oidc_client_id", class: "form-control", value: @data_holder.client_id, required: true
+    = error_messages_for(@data_holder, :client_id)
+    small.form-text.text-muted
+      | Client ID provided by OIDC Provider
+    
+  .form-group class="#{error_class_for(@data_holder, :client_secret)}"
+    = f.label :client_secret, "Secret"
+    = f.password_field :client_secret, id: "oidc_client_secret", class: "form-control", value: @data_holder.client_secret, required: true
+    = error_messages_for(@data_holder, :client_secret)
+    small.form-text.text-muted
+      | Secret associated with Relying Party's Client ID
```

Comment:
I think this is meant to say "Relaying" not "Relying"

Reply:
I truly appreciate someone looking closely enough to catch that. :)

While "relaying" would totally make sense, the app which federates authentication to an OIDC Provider (OP) is referred to as an RP, or "relying party".  I presume that's because it _relies_ upon information from the OP.  I was hesitant to use any of the "real" OIDC names, but it felt necessary here for precision; saying "application" or similar here seemed too vague.

Reference: https://openid.net/connect/faq/ (about halfway down; dang page doesn't use anchors on FAQ entries)

Reply:
Thanks for the extra information. I was not aware of that. Good job on using the correct terminology here and other places also.